doc/news/changes/minor/20231018SebastianKinnewig

```diff
@@ -0,0 +1,7 @@
+New: LinearAlgebra::TpetraWrappers::VectorReference for easier access 
+to LinearAlgebra::TpetraWrappers::Vector. Moreover, introducing several 
+new member functions to LinearAlgebra::TpetraWrappers::Vector, that 
+qualifies LinearAlgebra::TpetraWrappers::Vector to be used inside of 
+DynamicSparsityPattern.
```

Comment:
How is the `TpetraWrapers::Vector` used inside the `DynamicSparsityPattern`?

Reply:
I confused `DynamicSparsityPattern` with `AffineConstraints`. These changes add the necessary functions to TrilinosWrappers::Vector to use it as a "VectorType" in AffineConstraints.

However, it should be noted that TrilinosWrappers::Vector is still not being used in AffineConstraints because some TrilinosWrappers::SparseMatrix class is mandatory.
In my branch tpetra_wrappers, I already added a TpetraWrappers::SparseMatrix class and created a new instantiation for AffineConstraints with the following parameters:

```
    template void AffineConstraints<double>::distribute_local_to_global<
      LinearAlgebra::TpetraWrappers::SparseMatrix<double>,
      LinearAlgebra::TpetraWrappers::Vector<double>>(
      const FullMatrix<double> &,
      const Vector<double> &,
      const std::vector<AffineConstraints<double>::size_type> &,
      LinearAlgebra::TpetraWrappers::SparseMatrix<double> &,
      LinearAlgebra::TpetraWrappers::Vector<double> &,
      bool,
      std::integral_constant<bool, false>) const;
```

I will clarify this in this text and the pull request's title.